--- R-4.4.3.orig/R-4.4.3/src/library/base/R/lazyload.R	2024-03-27 07:02:03.000000000 +0800
+++ R-4.4.3/src/library/base/R/lazyload.R	2026-05-03 22:55:23.489194531 +0800
@@ -34,11 +34,9 @@
         .Internal(paste(list(...), sep, collapse, FALSE))
     readRDS <- function (file) {
         halt <- function (message) .Internal(stop(TRUE, message))
-        gzfile <- function (description, open)
-            .Internal(gzfile(description, open, "", 6))
         close <- function (con) .Internal(close(con, "rw"))
         if (! is.character(file)) halt("bad file name")
-        con <- gzfile(file, "rb")
+        con <- file(file, "rb")
         on.exit(close(con))
         .Internal(unserializeFromConn(con, baseenv()))
     }
