--- R-4.6.0.orig/src/library/tools/R/build.R	2026-03-27 07:02:15.000000000 +0800
+++ R-4.6.0/src/library/tools/R/build.R	2026-06-01 11:10:17.650096875 +0800
@@ -627,7 +627,7 @@
 	    partial <- db[containsBuildSexprs]
 	    dir.create("build", showWarnings = FALSE)
 	    ## version = 2L for maximal back-compatibility
-	    saveRDS(partial, build_partial_Rd_db_path, version = 2L)
+	    saveRDS(partial, build_partial_Rd_db_path, version = 2L, compress = FALSE)
 	}

         containsLaterSexprs <- which(btinfo[, "later"])
