--- R-4.4.3.orig/R-4.4.3/src/library/base/R/serialize.R	2024-03-27 07:02:03.000000000 +0800
+++ R-4.4.3/src/library/base/R/serialize.R	2026-05-03 22:41:10.296054843 +0800
@@ -26,12 +26,12 @@
 	object <- object # do not create corrupt file if object does not exist
 	mode <- if(ascii %in% FALSE) "wb" else "w"
 	con <- if (is.logical(compress))
-		   if(compress) gzfile(file, mode) else file(file, mode)
+		   file(file, mode)
 	       else
 		   switch(compress,
 			  "bzip2" = bzfile(file, mode),
 			  "xz"    = xzfile(file, mode),
-			  "gzip"  = gzfile(file, mode),
+			  "gzip"  = file(file, mode),
 			  stop("invalid 'compress' argument: ", compress))
         on.exit(close(con))
     }
@@ -48,7 +48,7 @@
 readRDS <- function(file, refhook = NULL)
 {
     if(is.character(file)) {
-        con <- gzfile(file, "rb")
+        con <- file(file, "rb")
         on.exit(close(con))
     } else if (inherits(file, "connection"))
 	con <- if(inherits(file, "url")) gzcon(file) else file
@@ -59,7 +59,7 @@
 infoRDS <- function(file)
 {
     if(is.character(file)) {
-        con <- gzfile(file, "rb")
+        con <- file(file, "rb")
         on.exit(close(con))
     } else if (inherits(file, "connection"))
         con <- if(inherits(file, "url")) gzcon(file) else file
